ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -6,68 +6,93 @@ import {
-    trackEvent({
-      category: MetaMetricsEventCategory.NotificationInteraction,
-      event: MetaMetricsEventName.NotificationClicked,
-      properties: {
-        notification_id: snapNotification.id,
-        notification_type: snapNotification.type,
-        previously_read: snapNotification.isRead,
-      },
-    });
-  };
-
-  const handleSnapButton = () => {
-    dispatch(markNotificationsAsRead([snapNotification.id]));
-    trackEvent({
-      category: MetaMetricsEventCategory.NotificationInteraction,
-      event: MetaMetricsEventName.NotificationClicked,
-      properties: {
-        notification_id: snapNotification.id,
-        notification_type: snapNotification.type,
-        previously_read: snapNotification.isRead,
-      },
-    });
-    history.push(getSnapRoute(snapNotification.data.origin));
-  };
-
-  return (
-    <NotificationListItemSnap
-      id={snapNotification.id}
-      snapId={snapNotification.data.origin}
-      isRead={snapNotification.isRead}
-      createdAt={new Date(snapNotification.createdAt)}
-      title={{
-        items: [
-          {
-            text: snapsNameGetter(snapNotification.data.origin) || 'Snap',
-          },
-        ],
-      }}
-      snapMessage={snapNotification.data.message}
-      handleSnapClick={handleSnapClick}
-      handleSnapButton={handleSnapButton}
-    />
-  );
+export const components: NotificationComponent<SnapNotification> = {
+  guardFn: isOfTypeNodeGuard(['snap' as typeof TRIGGER_TYPES.SNAP]),
+  item: ({ notification, onClick }) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const trackEvent = useContext(MetaMetricsContext);
+    const snapsMetadata = useSelector(getSnapsMetadata);
+    const snapsNameGetter = getSnapName(snapsMetadata);
+    const handleSnapButton = () => {
+      dispatch(markNotificationsAsRead([notification.id]));
+      trackEvent({
+        category: MetaMetricsEventCategory.NotificationInteraction,
+        event: MetaMetricsEventName.NotificationClicked,
+        properties: {
+          notification_id: notification.id,
+          notification_type: notification.type,
+          previously_read: notification.isRead,
+        },
+      });
+      history.push(getSnapRoute(notification.data.origin));
+    };
+    return (
+      <NotificationListItemSnap
+        id={notification.id}
+        snapId={notification.data.origin}
+        isRead={notification.isRead}
+        createdAt={new Date(notification.createdAt)}
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) || 'Snap',
+            },
+          ],
+        }}
+        snapMessage={notification.data.message}
+        handleSnapClick={onClick}
+        handleSnapButton={handleSnapButton}
+      />
+    );
+  },
+  details: {
+    title: ({ notification }) => (
+      <NotificationDetailTitle
+        title={notification.data.expandedView?.title as string}
```

Comment:
Is this property guaranteed to be set?

Reply:
yes, this is only ever called when the notification route is hit

Reply:
it is not

Reply:
Maybe I misunderstand, but it's optional here:

https://github.com/MetaMask/snaps/blob/f4890dd4eaa1b318f6fad00aa456e4600bc97225/packages/snaps-sdk/src/types/methods/notify.ts#L40-L43

Reply:
This component is only ever called in a notification detail route, the route is not navigated to unless there's a detailed view. The title is guaranteed to be there.